tests/testthat/_snaps/coerce.md

```diff
@@ -4,18 +4,25 @@
       expect_equal(coerce_chr(TRUE), "TRUE")
     Condition
       Warning:
-      Automatic coercion from type to character was deprecated in purrr 1.0.0.
+      Automatic coercion from logical to character was deprecated in purrr 1.0.0.
       i Please use an explicit call to as.character() within map_chr() instead.
     Code
       expect_equal(coerce_chr(1L), "1")
     Condition
       Warning:
-      Automatic coercion from type to character was deprecated in purrr 1.0.0.
+      Automatic coercion from integer to character was deprecated in purrr 1.0.0.
       i Please use an explicit call to as.character() within map_chr() instead.
     Code
       expect_equal(coerce_chr(1.5), "1.500000")
     Condition
       Warning:
-      Automatic coercion from type to character was deprecated in purrr 1.0.0.
+      Automatic coercion from double to character was deprecated in purrr 1.0.0.
       i Please use an explicit call to as.character() within map_chr() instead.
 
+# error captures correct env
+
+    Code
+      map_chr(1:4, identity)
```

Comment:
I'm clearly not passing on the correct environments in `deprecate_to_char()` — what am I doing wrong?

Reply:
The C function `caller_env()` implemented in purrr returns the current env. We could instead take the caller env from R (possibly with lazy assign) and pass that on through the C layer.